iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/historical/PipeHistoricalDataRegionTsFileAndDeletionExtractor.java

```diff
@@ -573,6 +611,27 @@ private boolean isTsFileGeneratedAfterExtractionTimeLowerBound(final TsFileResou
     }
   }
 
+  private void extractDeletions(
+      final DeletionResourceManager deletionResourceManager,
+      final List<PersistentResource> resourceList) {
+    LOGGER.info("Pipe {}@{}: start to extract deletions", pipeName, dataRegionId);
+    long startTime = System.currentTimeMillis();
+    List<DeletionResource> allDeletionResources = deletionResourceManager.getAllDeletionResources();
+    final int originalDeletionCount = allDeletionResources.size();
+    allDeletionResources =
+        allDeletionResources.stream()
+            .filter(resource -> !startIndex.isAfter(resource.getProgressIndex()))
+            .collect(Collectors.toList());
+    resourceList.addAll(allDeletionResources);
```

Comment:
Seemingly the resources' reference counts are not considered here? The deletion files may not be actually deleted at all after restart....

Reply:
BTW, all the followers, whether received or not, will receive the deletion after restart. It seems OK, but I'm not sure that if the improvement deserves to be done.

Reply:
Good catch! Resource's reference count should be considered here. fixed.